openest/generate/functions.py

```diff
@@ -372,10 +372,17 @@ def enable_deltamethod(self):
         for subcalc in self.subcalcs:
             subcalc.enable_deltamethod()
 
+    def partial_derivative(self, covariate, covarunit):
+        """
+        Returns a new calculation object that calculates the partial
+        derivative with respect to a given variable; currently only covariates are supported.
+        """
+        return Sum(map(lambda subcalc: subcalc.partial_derivative(covariate, covarunit), self.subcalcs))
+
     @staticmethod
     def describe():
         return dict(input_timerate='any', output_timerate='same',
-                    arguments=[arguments.calculationss],
+                    arguments=[arguments.calculationss, arguments.unshift.optional()],
```

Comment:
Might be a silly question but will adding this `arguments.unshift.optional()` back in (reverting 8fb62950997e7317c412cbeba2636c1b140f99d0) break other sectors/projections using `coefficient` functional forms?

Reply:
That's the whole point of the `calcchanges` branch-- it supports optional arguments, needed by energy, without breaking agriculture.